domain.c

```diff
@@ -215,12 +243,46 @@ void domain_maintain(void)
     force_tree_rebuild();
 }
 
+/* this function generates several domain decomposition policies for attempting
+ * creating the domain. */
+static int
+domain_policies_init(DomainDecompositionPolicy policies[],
+        const int NincreaseAlloc,
+        const int NdecreaseAccuracy,
+        const int SwitchToGlobal)
+{
+    int i;
+    for(i = 0; i < NincreaseAlloc; i ++) {
+        policies[i].TopNodeAllocFactor = All.TopNodeAllocFactor * pow(1.3, i);
+        policies[i].UseGlobalSort = All.DomainUseGlobalSorting;
+        policies[i].PreSort = 0;
+        policies[i].SubSampleDistance = 16;
+    }
+
+    for(i = NincreaseAlloc; i < NincreaseAlloc + NdecreaseAccuracy; i ++) {
+        policies[i].TopNodeAllocFactor = policies[NincreaseAlloc].TopNodeAllocFactor;
+        policies[i].UseGlobalSort = All.DomainUseGlobalSorting;
+        policies[i].PreSort = 0;
+        policies[i].SubSampleDistance = 16 * (i - NincreaseAlloc);
+    }
+
+    for(i = SwitchToGlobal; i < NincreaseAlloc + NdecreaseAccuracy; i ++) {
+        /* global sorting of particles is slow, so we add a presort to make the particle 
+         * distribution locally even to improve the balance */
+        policies[i].UseGlobalSort = 1;
+        policies[i].PreSort = 1;
```

Comment:
This adds a presort and global sort at the same time as decreasing the accuracy. Is that what you want?

Reply:
The comment above meant to say that since global sorting is slow anyway, a presort can always be added without making it much slower.